homeassistant/components/smartthings/climate.py

```diff
@@ -151,6 +158,24 @@ def _determine_features(self):
         # the entity state ahead of receiving the confirming push updates
         self.async_schedule_update_ha_state(True)
 
+    async def async_update(self):
+        """Update the attributes of the climate device."""
+        thermostat_mode = self._device.status.thermostat_mode
+        self._current_operation = MODE_TO_STATE.get(thermostat_mode)
+        if self._current_operation is None:
+            _LOGGER.debug('Device %s (%s) returned an invalid'
+                          'thermostat_mode: %s', self._device.label,
+                          self._device.device_id, thermostat_mode)
+
+        operation_mode_list = self._device.status.supported_thermostat_modes
+        if isinstance(operation_mode_list, Iterable):
+            self._operation_list = {MODE_TO_STATE.get(mode) for mode
```

Comment:
We're creating a set here, right?

Reply:
We don't want to include `None` as value in the list/set, right?

Reply:
Yes a `set` because there are multiple ST modes mapped to the same mode in HASS (i.e. `emergency heat` and `heat` both to go `heat`).

I tested with `None` in the operation list and it resulted in the front-end showing a blank option in the drop-down and when selected caused the service call to fail.  It's better to not do that and I've updated the code to log when it encounters an invalid option.